client/src/analysis/components/Clustering.tsx

```diff
@@ -0,0 +1,212 @@
+import * as React from "react";
+import { useState } from "react";
+import { useDispatch } from "react-redux";
+import { Accordion, Form, Icon } from "semantic-ui-react";
+import { defaultDebounce } from "../../helpers";
+import ResultList from "../../job/components/ResultList";
+import { AnalysisTypes } from "../../messages";
+import { cbToRadius, inRectConstraint, riConstraint, roConstraints } from "../../widgets/constraints";
+import DraggableHandle from "../../widgets/DraggableHandle";
+import Ring from "../../widgets/Ring";
+import { HandleRenderFunction } from "../../widgets/types";
+import * as analysisActions from "../actions";
+import { AnalysisProps } from "../types";
+import AnalysisLayoutTwoRes from "./AnalysisLayoutTwoRes";
+import useDefaultFrameView from "./DefaultFrameView";
+import { useRectROI } from "./RectROI";
+import Toolbar from "./Toolbar";
+
+
+
+
+const ClustAnalysis: React.SFC<AnalysisProps> = ({ analysis, dataset }) => {
+ 
+    const { shape } = dataset.params;
+    const [scanHeight, scanWidth, imageHeight, imageWidth] = shape;
+    const minLength = Math.min(imageWidth, imageHeight);
+
+    const [cx, setCx] = useState(imageWidth / 2);
+    const [cy, setCy] = useState(imageHeight / 2);
+    const [ri, setRi] = useState(minLength / 4);
+    const [ro, setRo] = useState(minLength / 2);
+
+    const riHandle = {
+        x: cx - ri,
+        y: cy,
+    }
+    const roHandle = {
+        x: cx - ro,
+        y: cy,
+    }
+
+    const [delta, setDelta] = React.useState(0);
```

Comment:
Is 0 the correct default value for `delta`? That makes the peak detection quite sensitive, right?